### PR TITLE
PC-1374: Link new Scottish EPC database columns to Supplier CSV

### PR DESCRIPTION
**Pull request by samyou-softwire**
[Link to Jira ticket](https://beisdigital.atlassian.net/browse/PC-1374)

# Description

exposes the data in scottish epc to the csv

does this by saving an object similar to an E&W csv to the session

to think about; is this fine or should we have a more stringent schema? I wasn't able to add one easily as
- I'd like to keep the format consistent across E&W & scotland
- the E&W layout cannot easily be made into a marshmallow schema as many keys have hyphens. hyphens cannot be a field name in python
- we could make a new schema, but then the E&W data would need to change format & the csv exporter would need more cases to handle

# Checklist

- [x] I have made any necessary updates to the documentation
- [x] I have checked there are no unnecessary IDE warnings in this PR
- [x] I have checked there are no unintentional line ending changes
- [x] I have added tests where applicable
- [x] If I have made any changes to the code, I have used the IDE auto-formatter on it
- [x] If I have made any changes to Python files (even if not changing any content strings), I have run `make extract-translations`
- [x] If I have made any changes to website flow, I have updated the [Flow Miro Board](https://miro.com/app/board/uXjVK4AgQ3E=/)
- [x] If I have made any changes to website flow, I have checked forward and back behaviour is still consistent

### CODE REVIEW

File: help_to_heat/frontdoor/interface.py [hunk 1]
```diff
@@ -525,14 +525,74 @@ def parse_dpa_to_address(self, dpa):
...
             epc = portal.models.ScottishEpcRating.objects.get(uprn=uprn)
         except portal.models.ScottishEpcRating.DoesNotExist:
             return {}
 
-        return {"uprn": epc.uprn, "rating": epc.rating, "date": epc.date}
+        improvements = epc.improvements or ""
+        alternative_improvements = epc.alternative_improvements or ""
+        all_improvements = improvements + (
+            "Alternatives: " + alternative_improvements if alternative_improvements else ""
+        )
```
Comment (jdgage) [inline help_to_heat/frontdoor/interface.py:539]: Do you need both fallbacks for the alternatives?

File: help_to_heat/frontdoor/views.py [hunk 2]
```diff
@@ -746,8 +746,7 @@ def save_post_data(self, data, session_id, page_name):
         if country == country_field_scotland and uprn is not None:
             epc = interface.api.epc.get_epc_scotland(uprn)
             if epc != {}:
-                epc_details = {"current-energy-rating": epc.get("rating"), "lodgement-date": epc.get("date")}
-                data[epc_details_field] = epc_details
```
Comment (jdgage) [inline help_to_heat/frontdoor/views.py:750]: Would prefer to just pass the fields we care about to the template, while we're changing this

**Review by jdgage — CHANGES_REQUESTED** [review]

File: help_to_heat/frontdoor/views.py (hunk 2 above)
Comment (samyou-softwire) [reply help_to_heat/frontdoor/views.py:750]: for E&W? in the scotland case here all the fields passed here are ones we care about

Comment (jdgage) [reply help_to_heat/frontdoor/views.py:750]: Ah sorry, I had this backwards - never mind

File: help_to_heat/frontdoor/interface.py [hunk 3]
```diff
@@ -525,14 +525,74 @@ def parse_dpa_to_address(self, dpa):
...
             epc = portal.models.ScottishEpcRating.objects.get(uprn=uprn)
         except portal.models.ScottishEpcRating.DoesNotExist:
             return {}
 
-        return {"uprn": epc.uprn, "rating": epc.rating, "date": epc.date}
+        improvements = epc.improvements
+        alternative_improvements = epc.alternative_improvements
+        all_improvements = improvements + (
+            "Alternatives: " + alternative_improvements if alternative_improvements else ""
+        )
```
Comment (jdgage) [inline help_to_heat/frontdoor/interface.py:539]: Two questions:
1. Should there be a space and/or some punctuation before 'Alternatives'?
2. Is `epc.improvements` ever going to be `None`? We'll need a fallback if so

Comment (samyou-softwire) [reply help_to_heat/frontdoor/interface.py:539]: It's not something I'm particularly concerned for, the load from the CSV means anything empty should become an empty string. If we want to account for it, then I'd probably want to `or ""` the other fields too

Comment (jdgage) [reply help_to_heat/frontdoor/interface.py:539]: I think throwing something in before 'Alternatives' would still be a good idea, so we don't have to make assumptions about the content of `epc.improvements`

Comment (samyou-softwire) [reply help_to_heat/frontdoor/interface.py]: Perhaps `|| Alternatives `? Like a separator but with more weight to it

**Review by jdgage — APPROVED** [review]

**Pull request merged**